openvpn: Allow override of interface name

miska opened the pull request:
If using a configuration file for OpenVPN, allow overriding name of the
interface. The reason is that then people could use configuration file
provided by VPN provider directly and override the name of the interface
to include it in correct firewall zone without need to alter the
configuration file.

Signed-off-by: Michal Hrusecky <michal@hrusecky.net>

miska committed:
openvpn: Allow override of interface name

If using a configuration file for OpenVPN, allow overriding name of the
interface. The reason is that then people could use configuration file
provided by VPN provider directly and override the name of the interface
to include it in correct firewall zone without need to alter the
configuration file.

Signed-off-by: Michal Hrusecky <michal@hrusecky.net>

jow- commented on package/network/services/openvpn/files/openvpn.init:
```diff
@@ -69,6 +69,38 @@ section_enabled() {
+
+	# If there is a no dev_type, try to guess it
+	if [ -z "$dev_type" ]; then
+		if grep -q '^[[:blank:]]*dev-type[[:blank:]]*tun' "$conf" || \
```
Option values in native configuration may also be single or double quoted. Maybe we can reuse https://github.com/openwrt/openwrt/blob/master/package/network/services/openvpn/files/etc/hotplug.d/openvpn/01-user#L3 and use a logic similar to this one:

```
local dev dev_type
if get_option dev dev || get_option dev_type dev-type; then
    case "${dev_type:-$dev}" in
        tun*) echo "--dev-type tun --dev $dev" ;;
        ?*) echo "--dev-type tap --dev $tap" ;;
    esac
fi

miska committed:
openpvn: Split out config parsing code for reuse

Split out code that parses openvpn configuration file into separate file
that can be later included in various scripts and reused.

Signed-off-by: Michal Hrusecky <michal@hrusecky.net>

miska commented:
What about like this?

jow- commented:
LGTM!

jow- commented:
Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/jow.git

jow- commented:
It is merged to master now.
